tests/functional/apps/eos.py

```diff
@@ -0,0 +1,216 @@
+        # response = public_key_len (1) ||
+        #            public_key (var) ||
+        #            address_len (1) ||
+        #            address (var) ||
+        #            chain_code (32)
+        offset: int = 0
+
+        public_key_len: int = response[offset]
+        offset += 1
+        public_key: bytes = response[offset:offset + public_key_len]
+        offset += public_key_len
+        address_len: int = response[offset]
+        offset += 1
+        address: str = response[offset:offset + address_len].decode("ascii")
+        offset += address_len
+        if request_chaincode:
+            chaincode: bytes = response[offset:offset + 32]
+            offset += 32
+        else:
+            chaincode = None
+
+        assert len(response) == offset
+        assert len(public_key) == 65
+        assert self.compute_adress_from_public_key(public_key) == address
+
+        return public_key, address, chaincode
+
+    def _get_public_key_params(self, derivation_path: bytes, confirm: bool, request_chaincode: bool):
+        p1 = P1_CONFIRM if confirm else P1_NON_CONFIRM
+        p2 = P2_CHAINCODE if request_chaincode else P2_NO_CHAINCODE
+        return CLA, INS.INS_GET_PUBLIC_KEY, p1, p2, derivation_path
+
+    def send_get_public_key_non_confirm(self, derivation_path: bytes, request_chaincode: bool) -> RAPDU:
+        apdu_params = self._get_public_key_params(derivation_path, False, request_chaincode)
+        return self._client.exchange(*apdu_params)
+
+    @contextmanager
+    def send_async_get_public_key_confirm(self, derivation_path: bytes, request_chaincode: bool):
+        apdu_params = self._get_public_key_params(derivation_path, True, request_chaincode)
+        with self._client.exchange_async(*apdu_params):
+            yield
+
+    def split_message(self, message: bytes) -> List[bytes]:
+        return [message[x:x + MAX_CHUNK_SIZE] for x in range(0, len(message), MAX_CHUNK_SIZE)]
+
+    def _send_sign_message(self, message: bytes, first: bool):
+        if first:
+            p1 = P1_FIRST
+        else:
+            p1 = P1_MORE
+        self._client.exchange(CLA, INS.INS_SIGN_MESSAGE, p1, 0, message)
+
+    @contextmanager
+    def _send_async_sign_message(self, message: bytes, first: bool):
+        if first:
+            p1 = P1_FIRST
+        else:
+            p1 = P1_MORE
+        with self._client.exchange_async(CLA, INS.INS_SIGN_MESSAGE, p1, 0, message):
+            yield
+
+    def send_async_sign_message(self,
+                                derivation_path: bytes,
+                                message: bytes) -> Generator[None, None, None]:
+        messages = self.split_message(derivation_path + message)
+        first = True
+
+        if len(messages) > 1:
+            self._send_sign_message(messages[0], True)
+            for m in messages[1:-1]:
+                self._send_sign_message_more(m, False)
```

Comment:
I can't find this `_send_sign_message_more()`

Reply:
yet tests are passing... not great.. fixed